lib/open.js

```diff
@@ -34,7 +34,7 @@ function makeArguments(filename, settings) {
     //   => "filename:1:2"
     //
     .replace(/\{filename\}(\S*)/, function(m, rest) {
-      return quote(info.filename + rest, settings.escapeQuotes);
+      return quote(info.filename, settings.escapeQuotes) + rest;
```

Comment:
Actually I suppose that we can split a `pattern` into the array of options and then apply the escaping algorithm to them one-by-one.  Here some examples how the pattern could be converted to the command line arguments:
`-g {filename}:{line}:{column}`  -> `-g "{filename}:{line}:{column}"` 
`-o {filename} -l {line} -c {column}` ->  `-o "{filename}" -l {line} -c {column}`
`vim {filename} "+call cursor({line}, {column})" -> `vim {filename} "+call cursor({line}, {column})"`
and so on.